kafka/pykafka/consumer.py

```diff
@@ -29,15 +133,63 @@ def topic(self):
     def partitions(self):
         return self._partitions
 
+    @property
+    def fetch_message_max_bytes(self):
+        return self._fetch_message_max_bytes
+
     def __iter__(self):
         while True:
-            yield self.consume()
+            message = self.consume(timeout=self._socket_timeout_ms)
+            if not message and self._consumer_timed_out():
+                raise StopIteration
+            yield message
+
+    def _consumer_timed_out(self):
+        if self._consumer_timeout_ms == -1:
+            return False
+        disp = (time.time() - self._last_message_time) * 1000.0
+        return disp > self._consumer_timeout_ms
```

Comment:
In cases where there are no messages waiting in any of the `OwnedPartition` queues, `Consumer.consume` will spin and continue to return `None`. Given this, I don't see a way to avoid tracking this timeout and still do something with the `consumer_timeout_ms` kwarg.